### PR TITLE
[Fix] Account filter in Process Payroll and Salary Component

### PR DESCRIPTION
**Pull request by KanchanChauhan**

### CODE REVIEW

File: erpnext/hr/doctype/salary_component/salary_component.js [hunk 1]
```diff
@@ -1,8 +1,13 @@
 // Copyright (c) 2016, Frappe Technologies Pvt. Ltd. and contributors
 // For license information, please see license.txt
 
-frappe.ui.form.on('Salary Component', {
-	refresh: function(frm) {
-
+cur_frm.set_query("default_account", "accounts", function(doc, cdt, cdn) {
```
Comment (rmehta) [inline erpnext/hr/doctype/salary_component/salary_component.js]: never use `cur_frm` !

add this in `setup` event

**Review by rmehta — CHANGES_REQUESTED** [review]

File: erpnext/hr/doctype/salary_component/salary_component.js (hunk 1 above)
Comment (KanchanChauhan) [inline erpnext/hr/doctype/salary_component/salary_component.js]: Done

**Review by rmehta — APPROVED** [review]

File: erpnext/hr/doctype/salary_component/salary_component.js [hunk 2]
```diff
@@ -2,7 +2,17 @@
 // For license information, please see license.txt
 
 frappe.ui.form.on('Salary Component', {
-	refresh: function(frm) {
-
+	setup: function(frm) {
+		frm.set_query("default_account", "accounts", function(doc, cdt, cdn) {
+			var d = locals[cdt][cdn];
+			var account_types = ["Bank", "Cash"];
```
Comment (nabinhait) [inline erpnext/hr/doctype/salary_component/salary_component.js]: Account type linked to salary company is not bank / cash, it should be Expense account (root_type="Expense")

**Review by nabinhait — CHANGES_REQUESTED** [review]

File: erpnext/hr/doctype/salary_component/salary_component.js (hunk 2 above)
Comment (KanchanChauhan) [inline erpnext/hr/doctype/salary_component/salary_component.js]: Added both account_type and root_type

**Review by nabinhait — APPROVED** [review]

**Pull request merged**